### ISSUE TITLE
My Home: Add go mobile primary task for native mobile devices

### ISSUE THREAD

**Issue by gwwar** [Automattic/wp-calypso#41880]
Let's add a new primary task for downloading the Native Mobile App. Note that Desktop views should never see this primary task. We'll render a tertiary card in that case. (This is already handled by the API).

- No illustration is necessary for this task
- When viewing with Android we see the android badge
- When viewing with an Apple device we see the apple badge.

| Native Mobile App Primary Task |  Desktop Tertiary Card |
| ------------- | ------------- |
| <img width="785" alt="81211889-0afc4400-8fa2-11ea-84af-7ef2aadb311a" src="https://user-images.githubusercontent.com/1270189/81215975-04b09c80-8f8f-11ea-92ba-355faabd57cd.png"> | <img width="1111" alt="Screen Shot 2020-05-06 at 10 14 35 AM" src="https://user-images.githubusercontent.com/1270189/81207456-6a4a5c00-8f82-11ea-9c8b-347b448f40db.png"> |

This issue has 3 parts:
- [x] Add a new primary task on the client in wp-calypso (#42020)
- [x] In the layout API, filter out the Go Mobile tertiary card to only render in Desktop views (D43094-code).
- [x] Add a new layout view with this task as it's only primary goal. Sites are only eligible to render this view if viewing with a native mobile device. (D43129-code)

**Comment by danhauk** [comment]
Here's some mockups of two different options that I'd welcome any feedback on. I'm leaning toward the app store badges because it adds the familiarity of the app store badges that people are used to seeing.

**Vanilla implementation**
This uses the exact same pattern and interaction already established with the other task cards. The button is primary and the "remind me" functions the same way.

<img width="389" alt="image" src="https://user-images.githubusercontent.com/448298/81211420-5a8e4000-8fa1-11ea-9d41-59c54cc00c58.png">

**App store badges**
This uses the same pattern, but replaces the primary button with the appropriate app store badge.

<img width="785" alt="image" src="https://user-images.githubusercontent.com/448298/81211889-0afc4400-8fa2-11ea-84af-7ef2aadb311a.png">

Any thoughts? cc @sixhours @jancavan @olesyabrk @lcollette

**Comment by gwwar** [comment]
It breaks the pattern with other tasks, but I do think showing the standard App store badges gives folks a better idea of where this will take them.

cc @elibud

**Comment by elibud** [comment]
+1 for showing the standard app store badges, thanks for the ping.

**Comment by jancavan** [comment]
+1 as well on using app store badges 👍

**Comment by gwwar** [comment]
Great unless folks say otherwise I think we have a winner with app store badges! This one should be ready to work on. Let me know if folks had questions on requirements and I can update the summary if it's unclear.

The client piece of this is up for grabs by anyone. If this is of any interest @Aurorum or @millerf add a comment on the issue.

**Comment by millerf** [comment]
Alright. I can give it a go...

**Comment by mmtr** [comment]
>  In the layout API, filter out the Go Mobile tertiary card to only render in Desktop views.

This should be covered in D43094-code

**Comment by millerf** [comment]
@gwwar I would need some docs on how to run the dev environment on mobile.
I tried last week with @Aurorum without success:
See [the bottom of my post here](https://github.com/Automattic/wp-calypso/pull/32822#issuecomment-623177450)

I guess there is an easy way but we couldn't find it...

**Comment by gwwar** [comment]
> I would need some docs on how to run the dev environment on mobile.

@millerf For this task, the device mode in devtools should be enough we basically just need to modify the UA string. Just remember to refresh the browser. FF and Safari also have similar modes if you prefer those for development.

https://developers.google.com/web/tools/chrome-devtools/device-mode

For bugs that can only be reproduced on a native device, local development for setup can be tricky. I normally use calypso.live links (which unfortunately doesn't run on forks), or end up spending quite a bit of time struggling with local dev environments if I need to step through.

**Comment by kwight** [comment]
This could roll out before we fully sort the "one-view-per-task" approach, so I've created D43129-code that adds a `FEATURE_GO_MOBILE_ON_MOBILE` card to the existing `SITE_ESTABLISHED` view.

**Comment by kwight** [comment]
@millerf FYI, server-side we're using `[jetpack_is_mobile](https://github.com/Automattic/jetpack/blob/8aeb14fe4fb4ce3241ce5e669e45a6087d3efa75/class.jetpack-user-agent.php#L13)` to determine whether the client is mobile or not (hence @gwwar 's suggestion of DevTools' device mode for testing) 👍

**Comment by gwwar** [comment]
I guess another thing to add is that we'll probably need to land the client component part of this first before any server updates. 

So @millerf feel free to hardcode render this new in primary as you work on this

**Comment by mmtr** [comment]
@millerf, if you want to expand a bit further @gwwar's suggestion, check out the testing instructions from this PR: https://github.com/Automattic/wp-calypso/pull/38905. They provide step-by-step instructions (including screenshots) to simulate a mobile device navigation in Chrome.

**Comment by millerf** [comment]
Hi @mmtr There shouldn't be any problem. I just haven't had the time to look into it yet. This week-end should be better..

**Comment by millerf** [comment]
@gwwar I have a couple of questions regarding this issue, especially with that is on wordpress.com actually:
 - There is already a "Get the wordpress  app" task:

| home | sub menu |
|-----|-----|
| ![Screenshot 2020-05-09 at 15 42 25](https://user-images.githubusercontent.com/790558/81475348-d7245700-920b-11ea-9937-ddedcfe371f7.png) | ![Screenshot 2020-05-09 at 15 42 35](https://user-images.githubusercontent.com/790558/81475349-d8558400-920b-11ea-8c0e-044ac9809174.png) |

 - There is already a "Go Mobile" at the end of the "home" as well:

![Screenshot 2020-05-09 at 15 45 05](https://user-images.githubusercontent.com/790558/81475436-89f4b500-920c-11ea-9bff-150a24c52048.png)

- In the new card design, what does the "remind me" link do?

- Should I reuse the [GoMobile](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/customer-home/cards/features/go-mobile/index.jsx#L54) component? I could modify it to fit the proposed card design. Or should I just create a new component?

**Comment by mmtr** [comment]
> There is already a "Get the wordpress app" task:
> What should I do with it? It seems redundant with the actual task...

Good catch. We might want to remove the task from the site setup list of tasks, but we can do that in a follow up. Or maybe we can ensure the task we're adding here only shows up if the user didn't downloaded it yet (this would need to be done server-side in D43129-code. cc @kwight).

> There is already a "Go Mobile" at the end of the "home" as well:

That's covered server-side in D43094-code. Once that lands, the Go Mobile card won't show up on mobile devices, while D43129-code ensures the new one is displayed.

> In the new card design, what does the "remind me" link do?

It will dismiss the task for a period of time. But that's already handled in the generic `Task` component, you shouldn't need to implement anything as long as you use that component:

https://github.com/Automattic/wp-calypso/blob/ecc977eee5ecceea69ed86d39840a5273814a7d3/client/my-sites/customer-home/cards/tasks/task.jsx#L176-L194

> Should I reuse the GoMobile component? I could modify it to fit the proposed card design. Or should I just create a new component?

All tasks should use the `Task` component as advised above so you can get most of the logic and design for free.

**Comment by kwight** [comment]
I closed this out today by deploying:
* #42020 
* #42127 
* D43129-code
* D43094-code

They were deployed in that order, as having the server-side patches live without the client PRs results in a WSOD (to keep in mind if reverting anything).
